api/client/samples/similar_regions/similar_region_state.py

```diff
@@ -88,7 +88,7 @@ def load(self):
         self._logger.info("Loading data")
         # Loop through the metric views...
         for name in self.region_properties:
-            path = os.path.join(CACHE_PATH, "{}.nbz".format(name))
+            path = os.path.join(os.path.dirname(__file__), CACHE_PATH, "{}.nbz".format(name))
```

Comment:
We don't want to force the path to be relative to the code base (e.g.  it might not be writeable by the user running the client). Making it relative to the current dir as now is ok, but better yet, how about using the tempfile lib.

Reply:
tempfile lib in python creates temporary directories/files during the lifetime of the python process. We want these to remain between invocations of similar regions so it's not the best approach. 

Earlier it was relative to the current dir, but then the problem is:
-if the cache directory doesn't exist, should we create it in whatever currdir the user's shell is in? I think that's a bad idea. 
-if the user invokes the file from one or more directories above, the cache directory now changes location. e.g. running 

``` python find_similar_region.py ...```  
vs
``` python similar_regions/find_similar_region.py``` 

will place and look for the cache directory at two different places.

Reply:
https://docs.python.org/3/distutils/setupscript.html#installing-package-data

This may be useful?